core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7882,31 +7925,34 @@ wasm_runtime_create_imports(WASMModuleCommon *module,
             continue;
         }
 
-        WASMExternInstance *extern_instance = imports + i;
+        WASMExternInstance *extern_instance = out + i;
         if (!wasm_runtime_create_extern_inst(module, import_type,
                                              extern_instance)) {
-            wasm_runtime_destroy_imports(module, imports);
+            wasm_runtime_destroy_imports(module, out);
```

Comment:
had better `memset(out, 0, sizeof(WASMExternInstance * import_count)` before L7917 first, or here it will traverse each extern instance in out array and free fields in extern instance if not NULL, and may free uninitiailized memory?